zapcore/error.go

```diff
@@ -37,6 +41,16 @@ import (
 // causer (from github.com/pkg/errors), a ${key}Causes field is added with an
 // array of objects containing the errors this error was comprised of.
 //
+// In the case of a ObjectMarshaler,
+//
+//	{
+//	  "error": {
```

Comment:
Technically this is a breaking change as we would have objects here instead of strings but I would argue that this would be the intended behavior when someone implements `ObjectMarshaler` on their error type.